compiler/test/dotty/tools/dotc/SettingsTests.scala

```diff
@@ -47,7 +47,7 @@ class SettingsTests {
 
     inContext {
       val args = List("-foo", "b", "-bar", "1")
-      val summary = Settings.processArguments(args, true)
+      val summary = Settings.processArguments(args, summon[Context].settingsState, true)
```

Comment:
You can use `ctx` instead of `summon[Context]` if you import `Contexts._` which contains `inline def ctx(using ctx: Context): Context = ctx`

Reply:
Actually, it was all wrong, because we used Seq of arguments from `ScalaSettings`, not our custom `Settings`. It was working well, becuase we were inserting values on positions 0 and 1, then we were retrieving them. Nevertheless, we have to use `Settings.defaultState` here, but we can actually make it default value, since most of the time we won't have semi initialized state.